addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -184,6 +187,16 @@ def write(self, vals):
     def unlink(self):
         # sudo: discuss.channel.rtc.session - cascade unlink of sessions for self member
         self.sudo().rtc_session_ids.unlink()  # ensure unlink overrides are applied
+        # if the only owner of the current channel is removed
+        for member in self.filtered(
+            lambda m: m == m.channel_id.channel_member_ids.filtered("is_owner")
+        ):
```

Comment:
This should only be done if there is no member left.

Reply:
Don't we want to make sure that there is always at least owner/admin in the channel?

Reply:
Yes but it should only be checked when the last owner is leaving, not "any" of them.

Reply:
`m == m.channel_id.channel_member_ids.filtered("is_owner")` means the selected member is the only owner. (it's `== `)